staramr/blast/JobHandler.py

```diff
@@ -93,8 +94,24 @@ def run_blasts(self, files) -> None:
         db_files = self._make_db_from_input_files(self._input_genomes_tmp_dir, files)
         logger.debug("Done making blast databases for input files")
 
-        logger.info("Scheduling MLST for input files")
-        self._schedule_mlst(db_files)
+        future_mlst_db = [] # type: list
+
+        if self._thread_pool_executor is not None:
+          logger.info("Scheduling MLST for input files")
+          future_mlst_db.append(self._thread_pool_executor.submit(self._schedule_mlst, db_files))
+
+        try:
```

Comment:
If you shift this `try/except` block after the `for file in db_files:` loop, then we will schedule both the MLST and BLAST jobs in the same thread pool in parallel rather than first waiting for all MLST jobs to finish before we schedule the BLAST jobs. This should speed things up a bit.

Could you try out this change?

Reply:
Yup, turns out it's a lot faster than having MLST run first than BLAST. Added this change for my next commit